### PR TITLE
Add test cases in Display & LVGL samples for boards with a supported display

### PR DESCRIPTION
**Pull request by ajarmouni-st**
This PR adds test cases in Display & LVGL samples for STM32 boards that come with a display (built-in or as a shield) that is
supported in Zephyr.
This patch will make it easier to catch display & LVGL regressions on STM32 boards in CI.

### CODE REVIEW

File: samples/subsys/display/lvgl/sample.yaml [hunk 1]
```diff
@@ -40,3 +40,36 @@ tests:
...
+    harness_config:
+      fixture: fixture_display
+    modules:
+      - lvgl
+    tags:
+      - samples
+      - display
+      - shield
+      - lvgl
+      - gui
```
Comment (faxe1008) [inline samples/subsys/display/lvgl/sample.yaml:75]: Can you maybe harmonize the tags across the entries? Some of them feature a `gui` tag some don't.

**Review by faxe1008 — DISMISSED** [review]

File: samples/subsys/display/lvgl/sample.yaml [hunk 2]
```diff
@@ -40,3 +40,36 @@ tests:
       - mimxrt595_evk_cm33
     integration_platforms:
       - mimxrt1170_evk_cm7
+  sample.subsys.display.lvgl.stm32:
```
Comment (erwango) [inline samples/subsys/display/lvgl/sample.yaml:43]: What's the difference with use case `sample.display.lvgl.gui` ?

File: samples/modules/lvgl/demos/sample.yaml [hunk 3]
```diff
@@ -17,3 +17,39 @@ tests:
   sample.modules.lvgl.demo_stress:
     extra_configs:
       - CONFIG_LV_Z_DEMO_STRESS=y
+  sample.modules.lvgl.demos.stm32:
```
Comment (erwango) [inline samples/modules/lvgl/demos/sample.yaml:20]: What is the added value vs common case ?

**Review by erwango — CHANGES_REQUESTED** [review]
Question on some use cases, as I'm not sure they require STM32 specific changes

**Comment by ajarmouni-st** [comment]
> What's the difference with use case `sample.display.lvgl.gui` ?

> What is the added value vs common case ?

@erwango STM32 specific test cases will force Twister to build the samples for the specified STM32 platforms in CI.

**Comment by ajarmouni-st** [comment]
Following @erwango's suggestion, modified/removed test cases that were needlessly STM32 specific.

**Review by kartben — DISMISSED** [review]

**Comment by ajarmouni-st** [comment]
Removed sample.subsys.display.lvgl.builtin test case since it's already covered by sample.display.lvgl.gui

**Review by erwango — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by faxe1008 — APPROVED** [review]

**Pull request merged**